Framework/DataHandling/src/LoadMask.cpp

```diff
@@ -39,6 +39,232 @@ using namespace Mantid::Kernel;
+  // 1. Split by space
+  std::vector<string> splitstrings;
+  boost::split(splitstrings, ins, boost::is_any_of(" "),
+               boost::token_compress_on);
+
+  // 2. Replace a-b to a - b, remove a-b and insert a, -, b
+  bool tocontinue = true;
+  size_t index = 0;
+  while (tocontinue) {
+    // a) Determine end of loop.  Note that loop size changes
+    if (index == splitstrings.size() - 1) {
+      tocontinue = false;
+    }
+
+    // b) Need to split?
+    vector<string> temps;
+    boost::split(temps, splitstrings[index], boost::is_any_of("-"),
+                 boost::token_compress_on);
+    if (splitstrings[index].compare("-") == 0 || temps.size() == 1) {
+      // Nothing to split
+      index++;
+    } else if (temps.size() == 2) {
+      // Has a '-' inside.  Delete and Replace
+      temps.insert(temps.begin() + 1, "-");
+      splitstrings.erase(splitstrings.begin() + index);
+      for (size_t ic = 0; ic < 3; ic++) {
+        if (temps[ic].size() > 0) {
+          splitstrings.insert(splitstrings.begin() + index, temps[ic]);
+          index++;
+        }
+      }
+    } else {
+      // Exception
+      std::string err = "String " + splitstrings[index] + " has too many '-'";
+      throw std::invalid_argument(err);
+    }
+
+    if (index >= splitstrings.size())
+      tocontinue = false;
+
+  } // END WHILE
+
+  // 3. Put to output integer vector
+  tocontinue = true;
+  index = 0;
+  while (tocontinue) {
+    // i)   push to the starting vector
+    ranges.push_back(
+        boost::lexical_cast<Mantid::specnum_t>(splitstrings[index]));
+
+    // ii)  push the ending vector
+    if (index == splitstrings.size() - 1 ||
+        splitstrings[index + 1].compare("-") != 0) {
+      // the next one is not '-'
+      ranges.push_back(
+          boost::lexical_cast<Mantid::specnum_t>(splitstrings[index]));
+      index++;
+    } else {
+      // the next one is '-', thus read '-', next
+      ranges.push_back(
+          boost::lexical_cast<Mantid::specnum_t>(splitstrings[index + 2]));
+      index += 3;
+    }
+
+    if (index >= splitstrings.size())
+      tocontinue = false;
+  } // END-WHILE
+}
+/*
+* Load and parse an ISIS masking file
+@param isisfilename :: the string containing full path to an ISIS mask file
+@param SpectraMasks :: output list of the spectra numbers to mask.
+*/
+void loadISISMaskFile(const std::string &isisfilename,
+                      std::vector<Mantid::specnum_t> &spectraMasks) {
+
+  std::vector<Mantid::specnum_t> ranges;
+
+  std::ifstream ifs;
+  ifs.open(isisfilename.c_str(), std::ios::in);
+  if (!ifs.is_open()) {
+    throw std::invalid_argument("Cannot open ISIS mask file" + isisfilename);
+  }
+
+  std::string isisline;
+  while (getline(ifs, isisline)) {
+    boost::trim(isisline);
+
+    // a. skip empty line
+    if (isisline.size() == 0)
+      continue;
+
+    // b. skip comment line
+    if (isisline.c_str()[0] < '0' || isisline.c_str()[0] > '9')
```

Comment:
that's scary -- it may slightly change the behaviour and I would better not to touch this remembering wide usage of this algorithm and variety of various mask files. 

I better leave it as it is.